### ISSUE TITLE
📝 HTML formatter inserts spaces after "{" even in whitespace sensitive contexts

### ISSUE THREAD

**Issue by terencehonles** [biomejs/biome#10241]
### Environment information

Biomejs Playground

### Configuration

```JSON

```

### Playground link

https://biomejs.dev/playground/?whitespaceSensitivity=strict&tab=formatter&pane=Diagnostics&code=PAAhAGQAbwBjAHQAeQBwAGUAIABoAHQAbQBsAD4ACgAgACAAPABoAHQAbQBsAD4ACgAgACAAIAAgADwAYgBvAGQAeQA%2BAAoAIAAgACAAIAA8AHMAcABhAG4APgB7AHQAZQBzAHQAfQAgAHsAewB0AGUAcwB0AH0AfQA8AC8AcwBwAGEAbgA%2BAAoAIAAgACAAIAA8AHAAcgBlAD4APABzAHAAYQBuAD4AewAlADwALwBzAHAAYQBuAD4APAAvAHAAcgBlAD4ACgAgACAAIAAgACAAIAA8AC8AYgBvAGQAeQA%2BAAoAIAAgADwALwBoAHQAbQBsAD4A&language=html

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by dyc3** [comment]
Can you elaborate more on your use case? what's your project like?

**Comment by dyc3** [comment]
Tentatively marking this as a bug anyway, because we shouldn't emit parser diagnostics for text that is obviously valid html.
